### ISSUE TITLE
Do not include satisfaction in policy id calculation

### ISSUE THREAD

**Issue by afilini** [bitcoindevkit/bdk#1218]
Currently every node in the `Policy` tree has an id which is calculated as a checksum on the string of the json-serialized object:

https://github.com/bitcoindevkit/bdk/blob/d6a0cf0795d9f8a1550e614e96b339a18fc537dc/crates/bdk/src/descriptor/policy.rs#L174-L178

The object includes details about whether the current wallet can satisfy the policy in some ways (aka `contribution`), and these are also included in the checksum. The same is true for what in a psbt is currently satisfied, which is the `satisfaction` field of `Policy`.

This causes the id of the nodes to change from wallet to wallet, depending on which private keys a wallet has or depending on which psbt you are looking at. Ideally if both wallets have the same public descriptor, they should also have the same id for the nodes, which makes sharing `policy_path`s much easier.

I would recommend changing the way the id is calculated, if you have any ideas let me know. I could also try working on it myself.

**Comment by notmandatory** [comment]
@matthiasdebernardini I believe your project uses this feature pretty heavily, any thoughts?

**Comment by LLFourn** [comment]
I was going to suggest simply removing the policy module in favor of pointing people to the planning module in miniscript. Thoughts?

**Comment by afilini** [comment]
Yeah that would be better, but as far as I know it still hasn't been released, right?

**Comment by LLFourn** [comment]
It's still not released yeah :\

**Comment by LLFourn** [comment]
Here you go:

https://github.com/rust-bitcoin/rust-miniscript/blob/c94deab5bd16b42da53ab38afcc521ae8c8bf024/src/plan.rs

It needs feedback and iterations but this is the way of the future.
